cask.el

```diff
@@ -806,6 +811,14 @@ url to the fetcher source."
         (setf (cask-dependency-ref dependency) ref))
       (-when-let (branch (plist-get args :branch))
         (setf (cask-dependency-branch dependency) branch)))
+    (-when-let (archive (plist-get args :archive))
+      (unless (boundp 'package-pinned-packages)
+        (signal 'cask-no-pinning nil))
+      (setq package-pinned-packages
```

Comment:
The function `add-to-list` is destructive, so you don't need to use `setq`. Also, the third option to `add-to-list` is `append`, you pass the list you are modifying.